### PR TITLE
Various bugfixes

### PR DESCRIPTION
**Pull request by summraznboi**
- Cleaned up @kevinfaveri used of spaced runes to be explicitly keyed with spacedRune, and fixed unit test
- Fixed bug where 2 etchings of the same rune in same block were not detecting each other

### CODE REVIEW

**Review by me-rob — APPROVED** [review]

File: index.ts [hunk 1]
```diff
@@ -118,12 +118,12 @@ export function encodeRunestoneUnsafe(runestone: RunestoneSpec): {
   }));
 
   let etching: Option<Etching> = None;
-  let etchingCommitment: string | undefined = undefined;
+  let etchingCommitment: Buffer | undefined = undefined;
```
Comment (mi-yu) [inline index.ts:121]: is this change related?

Comment (mi-yu) [reply index.ts:121]: oh nvm, i see that it shadows the input arg name

not related to this change, but we should choose a different name here to not conflict with the arg name for clarity

**Review by mi-yu — APPROVED** [review]

File: index.ts (hunk 1 above)
Comment (summraznboi) [reply index.ts:121]: Not arg name above, but the return type
